test/byContract/IexecPoco/IexecPoco1.test.ts

```diff
@@ -204,10 +231,219 @@ describe('IexecPoco1', () => {
+            const dealId = getDealId(iexecWrapper.getDomain(), orders.requester, taskIndex);
+            await expect(iexecPocoAsRequester.matchOrders(...orders.toArray())).to.emit(
+                iexecPoco,
+                'OrdersMatched',
+            );
+            // Check deal
+            const deal = await iexecPoco.viewDeal(dealId);
+            expect(deal.callback).to.equal(AddressZero);
+        });
+
+        it('[TEE] Should match orders without dataset', async () => {
+            orders.dataset.dataset = AddressZero;
+            orders.requester.dataset = AddressZero;
+            // Compute prices, stakes, rewards, ...
+            const dealPrice = (appPrice + workerpoolPrice) * volume; // no dataset price
+            const schedulerStake = await iexecWrapper.computeSchedulerDealStake(
+                workerpoolPrice,
+                volume,
+            );
+            // Deposit required amounts.
+            await iexecWrapper.depositInIexecAccount(requester, dealPrice);
+            await iexecWrapper.depositInIexecAccount(scheduler, schedulerStake);
+            // Sign and match orders.
+            await signOrders(iexecWrapper.getDomain(), orders, ordersActors);
+            const dealId = getDealId(iexecWrapper.getDomain(), orders.requester, taskIndex);
+            const tx = iexecPocoAsRequester.matchOrders(...orders.toArray());
+            // Check balances and frozen.
+            // Dataset price shouldn't be included.
+            await expect(tx).to.changeTokenBalances(
+                iexecPoco,
+                [iexecPoco, requester, scheduler],
+                [dealPrice + schedulerStake, -dealPrice, -schedulerStake],
+            );
+            expect(await iexecPoco.frozenOf(requester.address)).to.equal(dealPrice);
+            // Check events.
+            await expect(tx).to.emit(iexecPoco, 'OrdersMatched');
+            // Check deal
+            const deal = await iexecPoco.viewDeal(dealId);
+            expect(deal.dataset.pointer).to.equal(AddressZero);
+            expect(deal.dataset.owner).to.equal(AddressZero);
+            expect(deal.dataset.price.toNumber()).to.equal(0);
+            // Should not be impacted by datasetVolume == 0
+            expect(deal.botSize.toNumber()).to.equal(volume);
+        });
+
+        it('[TODO][TEE] Should match orders with restrictions', async () => {});
+
+        it('[TEE] Should fail when categories are different', async () => {
+            orders.requester.category = category + 1; // Valid but bad category.
+            await expect(iexecPocoAsRequester.matchOrders(...orders.toArray())).to.be.revertedWith(
+                'iExecV5-matchOrders-0x00',
+            );
+        });
+
+        it('[TEE] Should fail when category is unknown', async () => {
+            const categoriesCount = (await iexecPoco.countCategory()).toNumber();
+            orders.requester.category = categoriesCount + 1;
```

Comment:
I think there is not need for `+ 1` but maybe it can fools us to be more explicit ..!

Reply:
I changed it to make it clearer
```ts
            const lastCategoryIndex = (await iexecPoco.countCategory()).toNumber() - 1;
            orders.requester.category = lastCategoryIndex + 1;
            orders.workerpool.category = lastCategoryIndex + 1;
```